File: bulk/python/idor_01.py
# CWE-639: Insecure Direct Object Reference
from flask import Flask, request
app = Flask(__name__)
@app.route('/invoice/<int:invoice_id>')
def get_invoice(invoice_id):
    # No ownership check — any user can access any invoice
    return fetch_invoice(invoice_id)
def fetch_invoice(i): return str(i)

File: bulk/go/idor_01.go
// CWE-639: IDOR in HTTP handler
package main
import (
	"net/http"
	"fmt"
)
func getInvoice(w http.ResponseWriter, r *http.Request) {
	id := r.URL.Query().Get("id")
	// No ownership check — any authenticated user can access any invoice
	fmt.Fprintf(w, "invoice data for %s", id)
}
